### DISCUSSION TITLE
how to overwrite `tabBar` using `options` or `setOptions`

### DISCUSSION THREAD

**Discussion by ghasemikasra39** [react-navigation/react-navigation#10356] [Q&A]
Correct me if I am wrong, but I think as soon as we define a `tabBar` on `Tab.Navigator`, then there is no way to overwrite the `tabBar` with `options` defined on `Tab.Screen` or using `navigation.setOptions` on each screen:

```javascript
function BottomTabNavigator() {

  return (
      <Tab.Navigator
          initialRouteName={'Profile'}
          screenOptions={getScreenOptions}
          tabBar={props => <MyTabBar {...props}/>} // <== I have defined my own tabBar (MyTabBar)
      >
        <Tab.Screen name="Profile" 
           component={Profile} 
           options={{ tabBarActiveBackgroundColor: 'red' }} /> // <== tab-bar related options have NO EFFECT

        <Tab.Screen name="Explore" component={ExploreScreen} />
      </Tab.Navigator>
  );
}

const Explore = (props) => {

  useLayoutEffect(() => {
    const {navigation} = props
    
    const tabBarHandler = meta => {
      return <Text>Some text</Text>
    }

    navigation.setOptions({
      tabBarIcon: tabBarHandler, // <== tab-bar related options have NO EFFECT
    })
  }, [navigation])
}
```

Is there any solution for this scenario? (I **don't** want to remove  `tabBar` on `Tab.Navigator`. I know doing this will solve this scenario.)

### Why I need this? 
I have an screen (`Explore`) where I want to completely define a new set of tab buttons, that actually don't do any navigation. But instead, when they are pressed, they perform some other effects on the screen (And these new set of tab buttons must overwrite what `MyTabBar` show). And I **cannot** define these new set of tab buttons in my `MyTabBar` on `Tab.Navigator` because there (in the `MyTabBar`) I don't have access to the data I need (i.e. my scenario is completely depend on the state of `Explore` screen). Thid issue is so critical, because I cannot even hide the entire tab bar using `tabBarStyle: { display: 'none' }` in the `options` or `setOptions`.

**Comment by raajnadar** [comment]
First, if you want to disable navigation you can use the `tabPress` event handler to stop the navigation.

https://reactnavigation.org/docs/navigation-events/#listeners-prop-on-screen

Second, you can hide the tab bar with the `display: 'none'` property but it should apply correctly to the parent tab navigator.

If possible add a snack repro.
